firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionImpl.java

```diff
@@ -285,35 +296,39 @@ public UpdateTask updateApp() {
    * basic configuration and false for advanced configuration.
    */
   private UpdateTask updateApp(boolean showDownloadInNotificationManager) {
-    if (!isTesterSignedIn()) {
-      UpdateTaskImpl updateTask = new UpdateTaskImpl();
-      updateTask.setException(
-          new FirebaseAppDistributionException("Tester is not signed in", AUTHENTICATION_FAILURE));
-      return updateTask;
-    }
     return TaskUtils.onSuccessUpdateTask(
-        cachedNewRelease.get(),
+        signInStorage.getSignInStatus(),
         lightweightExecutor,
-        release -> {
-          if (release == null) {
-            LogWrapper.v(TAG, "New release not found.");
+        signedIn -> {
+          if (!signedIn) {
             return getErrorUpdateTask(
                 new FirebaseAppDistributionException(
-                    ErrorMessages.RELEASE_NOT_FOUND_ERROR, UPDATE_NOT_AVAILABLE));
-          }
-          if (release.getDownloadUrl() == null) {
-            LogWrapper.v(TAG, "Download failed to execute.");
-            return getErrorUpdateTask(
-                new FirebaseAppDistributionException(
-                    ErrorMessages.DOWNLOAD_URL_NOT_FOUND,
-                    FirebaseAppDistributionException.Status.DOWNLOAD_FAILURE));
-          }
-
-          if (release.getBinaryType() == BinaryType.AAB) {
-            return aabUpdater.updateAab(release);
-          } else {
-            return apkUpdater.updateApk(release, showDownloadInNotificationManager);
+                    "Tester is not signed in", AUTHENTICATION_FAILURE));
```

Comment:
Can this use `assertTesterIsSignedIn()`?

Reply:
Oh actually yes! It's a little tricky because I have to use `TaskUtils.onSuccessUpdateTask()` but it works.